src/daos-deserialize/daos-deserialize.c

```diff
@@ -0,0 +1,225 @@
+    uuid_t pool_uuid;
+
+    int usage = 0;
+    while(1) {
+        int c = getopt_long(
+                    argc, argv, "p:vh",
+                    long_options, &option_index
+                );
+
+        if (c == -1) {
+            break;
+        }
+
+        switch(c) {
+            case 'p':
+                uuid_parse(optarg, pool_uuid);
+                break;
+            case 'v':
+                mfu_debug_level = MFU_LOG_VERBOSE;
+                break;
+            case 'h':
+                usage = 1;
+                break;
+            case '?':
+                usage = 1;
+                break;
+            default:
+                if(rank == 0) {
+                    printf("?? getopt returned character code 0%o ??\n", c);
+                }
+        }
+    }
+
+    /* If we need to print the usage
+     * then do so before internal processing */
+    if (usage) {
+        if (rank == 0) {
+            print_usage();
+        }
+        mfu_finalize();
+        MPI_Finalize();
+        return 1;
+    }
+
+    char** argpaths = (&argv[optind]);
+    
+    /* The remaining arguments are treated as src/dst paths */
+    int numpaths = argc - optind;
+
+    /* advance to next set of options */
+    optind += numpaths;
+
+    /* Before processing, make sure we have at least two paths to begin with */
+    if (numpaths < 1) {
+        MFU_LOG(MFU_LOG_ERR, "At least one pool and container is required:"
+                MFU_ERRF, MFU_ERRP(-MFU_ERR_INVAL_ARG));
+        mfu_finalize();
+        MPI_Finalize();
+        return 1;
+    }
+
+    /* TODO: support input file rather than required all pool/cont paths
+     * be input on cmd line */
+    /* TODO: maybe support moving into multiple pools? */
+    /* TODO: broadcast pool handle to all ranks if not supporting
+     * multiple pools (instead of connecting on each rank)
+     * or have a check that if all pools are
+     * the same, then use the same handle */
+
+    /* create an empty file list */
+    mfu_flist tmplist = mfu_flist_new();
+
+    rc = daos_init();
+    if (rc != 0) {
+        MFU_LOG(MFU_LOG_ERR, "Failed to initialize daos");
+        return 1;
+    }
+
+    /* use rank 0's paths, and spread them evenly among ranks
+     * Each "path" is a DAOS pool/cont, and this tool will
+     * serialize each container into its own hdf5 file */
+    int i;
+    if (rank == 0) {
+        for (i = 0; i < numpaths; i++) {
+                uint64_t idx = mfu_flist_file_create(tmplist);
+                mfu_flist_file_set_cont(tmplist, idx, argpaths[i]);
+        }
+    }
+
+    mfu_flist_summarize(tmplist);
+    mfu_flist newflist = mfu_flist_spread(tmplist);
+
+    /* get size of local list for each rank */
+    uint64_t size = mfu_flist_size(newflist);
+
+    daos_handle_t poh;
+    uuid_t cont_uuids[size];
+    daos_handle_t cohs[size];
+    bool daos_no_prefix = true;
+    daos_pool_info_t pool_info = {0};
+
+#if DAOS_API_VERSION_MAJOR < 1
```

Comment:
@dsikich I'm not 100% sure, but we may be able to remove the DAOS_API_VERSION_MAJOR  check now